client/my-sites/site-settings/speed-up-site-settings.jsx

```diff
@@ -14,23 +14,63 @@ import { connect } from 'react-redux';
 import Card from 'components/card';
 import JetpackModuleToggle from 'my-sites/site-settings/jetpack-module-toggle';
 import FormFieldset from 'components/forms/form-fieldset';
+import CompactFormToggle from 'components/forms/form-toggle/compact';
+import FormSettingExplanation from 'components/forms/form-setting-explanation';
 import isJetpackModuleUnavailableInDevelopmentMode from 'state/selectors/is-jetpack-module-unavailable-in-development-mode';
 import isJetpackSiteInDevelopmentMode from 'state/selectors/is-jetpack-site-in-development-mode';
+import { activateModule, deactivateModule } from 'state/jetpack/modules/actions';
+import isJetpackModuleActive from 'state/selectors/is-jetpack-module-active';
+import isActivatingJetpackModule from 'state/selectors/is-activating-jetpack-module';
+import isDeactivatingJetpackModule from 'state/selectors/is-deactivating-jetpack-module';
 import { getSelectedSiteId } from 'state/ui/selectors';
 import { getSiteSlug } from 'state/sites/selectors';
 import SupportInfo from 'components/support-info';
 
 class SpeedUpSiteSettings extends Component {
+	static defaultProps = {
+		togglingSiteAccelerator: false,
+	};
+
 	static propTypes = {
 		fields: PropTypes.object,
 		isRequestingSettings: PropTypes.bool,
 		isSavingSettings: PropTypes.bool,
 		jetpackVersionSupportsLazyImages: PropTypes.bool,
+		siteAcceleratorSupported: PropTypes.bool,
+		activateModule: PropTypes.func,
+		deactivateModule: PropTypes.func,
 
 		// Connected props
 		photonModuleUnavailable: PropTypes.bool,
 		selectedSiteId: PropTypes.number,
 		siteSlug: PropTypes.string,
+		photonModuleActive: PropTypes.bool,
+		assetCdnModuleActive: PropTypes.bool,
+		togglingSiteAccelerator: PropTypes.bool,
+	};
+
+	handleCdnChange = () => {
+		const { photonModuleActive, assetCdnModuleActive, selectedSiteId } = this.props;
+
+		// Check if any of the CDN options are on.
+		const cdnStatus = photonModuleActive || assetCdnModuleActive;
+
+		// If one of them is on, we turn everything off.
+		if ( true === cdnStatus ) {
```

Comment:
Also, if we reduce the level of specificity (which I'm not sure we need), we could go with something like:

```
handleCdnChange = () => {
	const {
		activateModule,
		assetCdnModuleActive,
		deactivateModule,
		photonModuleActive,
		selectedSiteId,
	} = this.props;
		
	// If one of them is on, we turn everything off.
	if ( photonModuleActive || assetCdnModuleActive ) {
		deactivateModule( selectedSiteId, 'photon', true );
		deactivateModule( selectedSiteId, 'photon-cdn', true );
	} else {
		activateModule( selectedSiteId, 'photon', true );
		activateModule( selectedSiteId, 'photon-cdn', true );
	}
}
```

Reply:
If I do this, I think we'll end up making unnecessary requests if some cases.

Consider for example a case where you have only one option out of the 2 active. If you toggle the master toggle, you'll end up doing `deactivateModule` for the 2 modules, even if one of them does not need to be deactivated because it is already inactive.

I've tried to simplify things a bit though. 👍

Reply:
I see, thanks for elaborating - guess we need a better way to structure that logic. Because just imagine if we want to include a third module, or a fourth module in the logic - the logic would then grow indefinitely.